packages/shared_preferences/shared_preferences/lib/shared_preference_async.dart

```diff
@@ -0,0 +1,313 @@
+          !_cacheOptions.filter.allowList!.contains(element));
+    }
+    return keys;
+  }
+
+  /// Reads a value of any type from the cache.
+  Object? get(String key) => _cache[key];
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not a
+  /// bool.
+  bool? getBool(String key) => get(key) as bool?;
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not
+  /// an int.
+  int? getInt(String key) => get(key) as int?;
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not a
+  /// double.
+  double? getDouble(String key) => get(key) as double?;
+
+  /// Reads a value from the cache, throwing a [TypeError] if the value is not a
+  /// String.
+  String? getString(String key) => get(key) as String?;
+
+  /// Reads a set of string values from the cache, throwing an
+  /// exception if it's not a string list.
+  List<String>? getStringList(String key) {
+    List<dynamic>? list = _cache[key] as List<dynamic>?;
+    list = list?.cast<String>();
+    if (list is List<String>) {
+      _cache[key] = list.toList();
+    }
+    // Make a copy of the list so that later mutations won't propagate
+    return list?.toList() as List<String>?;
+  }
+
+  /// Saves a boolean [value] to the cache and platform.
+  Future<bool> setBool(String key, bool value) async {
+    _cache[key] = value;
+    return _directAccess.setBool(key, value);
+  }
+
+  /// Saves an integer [value] to the cache and platform.
+  Future<bool> setInt(String key, int value) async {
+    _cache[key] = value;
+    return _directAccess.setInt(key, value);
+  }
+
+  /// Saves a double [value] to the cache and platform.
+  ///
+  /// On platforms that do not support storing doubles,
+  /// the value will be stored as a float instead.
+  Future<bool> setDouble(String key, double value) async {
+    _cache[key] = value;
+    return _directAccess.setDouble(key, value);
+  }
+
+  /// Saves a string [value] to the cache and platform.
+  ///
+  /// Note: Due to limitations on some platforms,
+  /// values cannot start with the following:
+  ///
+  /// - 'VGhpcyBpcyB0aGUgcHJlZml4IGZvciBhIGxpc3Qu'
+  /// - 'VGhpcyBpcyB0aGUgcHJlZml4IGZvciBEb3VibGUu'
+  Future<bool> setString(String key, String value) async {
+    _cache[key] = value;
+    return _directAccess.setString(key, value);
+  }
+
+  /// Saves a list of strings [value] to the cache and platform.
+  Future<bool> setStringList(String key, List<String> value) async {
+    _cache[key] = value;
+    return _directAccess.setStringList(key, value);
+  }
+
+  /// Removes an entry from cache and platform.
+  Future<bool> remove(String key) async {
+    _cache.remove(key);
+    return _directAccess.remove(key);
+  }
+
+  /// Clears cache and platform preferences that match filter options.
+  Future<bool> clear() async {
+    if (_cacheOptions.filter.allowList == null) {
+      _cache.clear();
+    } else {
+      _cache.removeWhere(
```

Comment:
Since the options are immutable, can't we always `_cache.clear()`, since the only things in the cache should be things that match the filter?